packages/zowe-explorer/src/trees/dataset/DatasetActions.ts

```diff
@@ -1079,6 +1079,109 @@ export class DatasetActions {
         }
     }
 
+    /**
+     * Attempts to open a data set from the selection of text made in the editor
+     * Works the same as ZOOM command in TSO/ISPF
+     *
+     */
+    public static async zoom(): Promise<void> {
+        ZoweLogger.trace("dataset.actions.zoom called.");
+
+        const editor = vscode.window.activeTextEditor;
+        if (!editor) {
+            const notActiveEditorMsg = vscode.l10n.t("No active editor open. Please open a file and select text to open a data set.");
+            Gui.warningMessage(notActiveEditorMsg);
+            return;
+        }
+
+        const doc = editor.document;
+        const selection = editor.selection;
+        const selectedText = doc.getText(selection);
+        // Shouldn't happen but just in case
+        if (!selectedText) {
+            Gui.warningMessage(vscode.l10n.t("No selection to open."));
+            return;
+        }
+
+        // If selected text is not a valid data set name or data set(member name)
+        const ds = DatasetUtils.extractDataSetAndMember(selectedText);
+        const hasMember = ds.memberName && ds.memberName.length > 0;
+        if (hasMember) {
+            if (!DatasetUtils.validateMemberName(ds.memberName)) {
+                Gui.warningMessage(vscode.l10n.t("Selection is not a valid data set member name."));
+                return;
+            }
+        } else if (!DatasetUtils.validateDataSetName(ds.dataSetName)) {
+            Gui.warningMessage(vscode.l10n.t("Selection is not a valid data set name."));
+            return;
+        }
+
+        let sessProfileName: string = "";
+        const profiles = Profiles.getInstance();
+        const profileNamesList = ProfileManagement.getRegisteredProfileNameList(Definitions.Trees.MVS);
+        if (doc.uri && FsAbstractUtils.getInfoForUri(doc.uri)?.profileName) {
+            sessProfileName = FsAbstractUtils.getInfoForUri(doc.uri).profileName;
+        }
```

Comment:
We can join the declaration and assignment of `sessProfileName` before the remaining checks:

```suggestion
        const profiles = Profiles.getInstance();
        const profileNamesList = ProfileManagement.getRegisteredProfileNameList(Definitions.Trees.MVS);
        let sessProfileName = doc.uri ? FsAbstractUtils.getInfoForUri(doc.uri)?.profileName : "";
```